edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -476,6 +493,33 @@ async Task HandleNewServiceIdentity(ServiceIdentity serviceIdentity)
             }
         }
 
+        Task<SynchedPurchase> GetPurchaseAsync(string id, Option<PurchaseContent> purchase, bool refresh = false)
+        {
+            var lastSingleIdentityRefresh = this.identitiesLastRefreshTime.ContainsKey(id) ? this.identitiesLastRefreshTime[id] : this.cacheLastRefreshTime;
+            var lastSynched = lastSingleIdentityRefresh > this.cacheLastRefreshTime ? lastSingleIdentityRefresh : this.cacheLastRefreshTime;
+
+            return purchase.Match(
+                p => Task.FromResult(new SynchedPurchase(lastSynched, purchase)),
+                async () =>
+                {
+                    // TODO: Remove full cache refresh when IotHub APi is fixed
```

Comment:
NP: Create a PBI and link it to the comment in the code. Comments can easily be forgotten.